Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs

```diff
@@ -89,6 +89,18 @@ public override void ViewDidDisappear(bool animated)
 
 			_appeared = false;
 			PageController.SendDisappearing();
+
+			var navigationPage = Element.Parent as NavigationPage;
+			if (navigationPage == null)
+				return;
+
+			int managedStackCount = navigationPage.Navigation.NavigationStack.Count;
+			IVisualElementRenderer renderer = Platform.GetRenderer(navigationPage);
+			var navigationController = renderer as UINavigationController;
+
+			int? nativeStackCount = navigationController?.ViewControllers.Length;
```

Comment:
it's probably cheaper, and more clear to wrap this in a `if (navigationController != null)` than playing with `int?`

like this:
```
if (navigationController != null && navigationController.ViewControllers.Length <= managedStackCount)
    await your stuff;
```